client/src/actions/collection.js

```diff
@@ -94,6 +94,23 @@ export function addDatasetToCollection(datasetId, collectionId) {
   };
 }
 
+export function addVisualisationToCollection(datasetId, collectionId) {
+  return (dispatch, getState) => {
+    const collection = getState().library.collections[collectionId];
+    collection.visualisations.push(datasetId);
+    dispatch(editCollection(collection));
+    dispatch(showNotification('info', `Added to ${collection.title}`, true));
+  };
+}
+export function addDashboardToCollection(datasetId, collectionId) {
```

Comment:
change parameter name to `dashboardId`

Reply:
now that we have 3 copies of the same function, the only part that changes is
```
collection.<entityType>.push(entityId)
```
Does it make sense to refactor that and create a single function:

```js
addEntityToCollection(entityType, entityId, collectionId) {
 return (dispatch, getState) => {
    const collection = getState().library.collections[collectionId];
    collection[entityType + "s"].push(datasetId);
    dispatch(editCollection(collection));
    dispatch(showNotification('info', `Added to ${collection.title}`, true));
  };
}
```

When using it

```js
dispatch(addEntityToCollection("dataset", dataset.id, collectionId))